planner/core/errors.go

```diff
@@ -19,74 +19,73 @@ import (
+	codePartitionClauseOnNonpartitioned                = mysql.ErrPartitionClauseOnNonpartitioned
+	codeDBaccessDenied                                 = mysql.ErrDBaccessDenied
+	codeTableaccessDenied                              = mysql.ErrTableaccessDenied
+	codeSpecificAccessDenied                           = mysql.ErrSpecificAccessDenied
+	codeViewNoExplain                                  = mysql.ErrViewNoExplain
+	codeWindowFrameStartIllegal                        = mysql.ErrWindowFrameStartIllegal
+	codeWindowFrameEndIllegal                          = mysql.ErrWindowFrameEndIllegal
+	codeWindowFrameIllegal                             = mysql.ErrWindowFrameIllegal
+	codeWindowRangeFrameOrderType                      = mysql.ErrWindowRangeFrameOrderType
+	codeWindowRangeFrameTemporalType                   = mysql.ErrWindowRangeFrameTemporalType
+	codeWindowRangeFrameNumericType                    = mysql.ErrWindowRangeFrameNumericType
+	codeWindowRangeBoundNotConstant                    = mysql.ErrWindowRangeBoundNotConstant
+	codeWindowRowsIntervalUse                          = mysql.ErrWindowRowsIntervalUse
+	codeWindowFunctionIgnoresFrame                     = mysql.ErrWindowFunctionIgnoresFrame
+	codeUnsupportedOnGeneratedColumn                   = mysql.ErrUnsupportedOnGeneratedColumn
 )
 
 // error definitions.
 var (
-	ErrUnsupportedType             = terror.ClassOptimizer.New(codeUnsupportedType, "Unsupported type %T")
-	ErrAnalyzeMissIndex            = terror.ClassOptimizer.New(codeAnalyzeMissIndex, "Index '%s' in field list does not exist in table '%s'")
-	ErrCartesianProductUnsupported = terror.ClassOptimizer.New(codeUnsupported, "Cartesian product is unsupported")
-	ErrStmtNotFound                = terror.ClassOptimizer.New(codeStmtNotFound, "Prepared statement not found")
-	ErrWrongParamCount             = terror.ClassOptimizer.New(codeWrongParamCount, "Wrong parameter count")
-	ErrSchemaChanged               = terror.ClassOptimizer.New(codeSchemaChanged, "Schema has changed")
+	ErrUnsupportedType             = terror.ClassOptimizer.New(codeUnsupportedType, mysql.MySQLErrName[mysql.ErrUnsupportedType])
+	ErrAnalyzeMissIndex            = terror.ClassOptimizer.New(codeAnalyzeMissIndex, mysql.MySQLErrName[mysql.ErrAnalyzeMissIndex])
+	ErrCartesianProductUnsupported = terror.ClassOptimizer.New(codeUnsupported, mysql.MySQLErrName[mysql.ErrCartesianProductUnsupported])
+	ErrStmtNotFound                = terror.ClassOptimizer.New(codeStmtNotFound, mysql.MySQLErrName[mysql.ErrPreparedStmtNotFound])
+	ErrWrongParamCount             = terror.ClassOptimizer.New(codeWrongParamCount, mysql.MySQLErrName[mysql.ErrWrongParamCount])
+	ErrSchemaChanged               = terror.ClassOptimizer.New(codeSchemaChanged, mysql.MySQLErrName[mysql.ErrSchemaChanged])
 	ErrTablenameNotAllowedHere     = terror.ClassOptimizer.New(codeTablenameNotAllowedHere, "Table '%s' from one of the %ss cannot be used in %s")
```

Comment:
We should use the predefined error message.

Reply:
This one is a little different, it has a predefined error message in `mysql/errname.go` for `ErrTablenameNotAllowedHere`, but it is different from the native error message.

```
	ErrTablenameNotAllowedHere:                  "Table '%-.192s' from one of the SELECTs cannot be used in %-.32s",
```

One work around is to then create another error code for this error, say `ErrTablenameNotAllowedHere2` if we want every error message to be strictly defined in `mysql/errname.go`.

Reply:
I think we could modify the error message in  mysql/errname.go